packaging/macos/uninstall.sh

```
@@ -0,0 +1,225 @@
+# ---- preflight ----------------------------------------------------------
+
+[[ "$(uname -s)" == "Darwin" ]] || die "macOS only"
+[[ $EUID -eq 0 ]] || die "must run as root (try: sudo $0 $*)"
+
+if [[ "${PURGE}" == "true" ]]; then
+  if [[ -z "${TARGET_USER}" ]]; then
+    TARGET_USER="${SUDO_USER:-}"
+  fi
+  if [[ -n "${TARGET_USER}" ]]; then
+    TARGET_HOME="$(dscl . -read "/Users/${TARGET_USER}" NFSHomeDirectory 2>/dev/null | awk '{print $2}')"
+    TARGET_UID="$(id -u "${TARGET_USER}" 2>/dev/null || echo "")"
```

Shellcheck (0.11.0)

[warning] TARGET_UID appears unused. Verify use (or export if used externally).

(SC2034)